server/src/main/java/org/opensearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -100,20 +105,30 @@ public AllocationService(
         ClusterInfoService clusterInfoService,
         SnapshotsInfoService snapshotsInfoService
     ) {
-        this(allocationDeciders, shardsAllocator, clusterInfoService, snapshotsInfoService);
+        this(allocationDeciders, shardsAllocator, clusterInfoService, snapshotsInfoService, NoopMetricsRegistry.INSTANCE);
         setExistingShardsAllocators(Collections.singletonMap(GatewayAllocator.ALLOCATOR_NAME, gatewayAllocator));
     }
 
     public AllocationService(
         AllocationDeciders allocationDeciders,
         ShardsAllocator shardsAllocator,
         ClusterInfoService clusterInfoService,
-        SnapshotsInfoService snapshotsInfoService
+        SnapshotsInfoService snapshotsInfoService,
+        MetricsRegistry metricsRegistry
     ) {
         this.allocationDeciders = allocationDeciders;
         this.shardsAllocator = shardsAllocator;
         this.clusterInfoService = clusterInfoService;
         this.snapshotsInfoService = snapshotsInfoService;
+        initializeMetrics(metricsRegistry);
+    }
+
+    private void initializeMetrics(MetricsRegistry metricsRegistry) {
+        this.rerouteHistogram = metricsRegistry.createHistogram(
```

Comment:
@Gaganjuneja are we considering to use annotations to use boilerplate code this inline latency metrics if those can be achieved by annotations and using method name for emitting the metrics?

Reply:
I agree with you, we need less invasive way. We can go ahead with this PR now and will open an another issue for that and as part of that feature we will migrate all the existing metrics.

Reply:
@Gaganjuneja this is important otherwise intrumentation code is sprinkled all over the classes. Are you planning to prioritize this?

Reply:
Resolving this conversation, since @Gaganjuneja will be following up on this on a separate issue.